tokio-util/src/io/sync_bridge.rs

```diff
@@ -6,6 +6,255 @@ use tokio::io::{
 
 /// Use a [`tokio::io::AsyncRead`] synchronously as a [`std::io::Read`] or
 /// a [`tokio::io::AsyncWrite`] as a [`std::io::Write`].
+///
+/// # Alternatives
+///
+/// In many cases, there are better alternatives to using `SyncIoBridge`, especially
+/// if you want to avoid blocking the async runtime. Consider the following scenarios:
+///
+/// When hashing data, using `SyncIoBridge` can lead to suboptimal performance and
+/// might not fully leverage the async capabilities of the system.
+///
+/// ### Why It Matters:
+///
+/// `SyncIoBridge` allows you to use synchronous I/O operations in an asynchronous
+/// context by blocking the current thread. However, this can be inefficient because:
+/// - **Blocking**: The use of `SyncIoBridge` may block a valuable async runtime
+///   thread, which could otherwise be used to handle more tasks concurrently.
+/// - **Thread Pool Saturation**: If many threads are blocked using `SyncIoBridge`,
+///   it can exhaust the async runtime's thread pool, leading to increased latency
+///   and reduced throughput.
+/// - **Lack of Parallelism**: By blocking on synchronous operations, you may miss
+///   out on the benefits of running tasks concurrently, especially in I/O-bound
+///   operations where async tasks could be interleaved.
```

Comment:
It can't block a valuable async runtime thread. Attempting to use it on a runtime thread results in a panic. I would just say that it takes up an entire OS thread, which is inefficient compared to what is usually possible with async code, which can run many things on one thread.